src/core/ddsc/src/dds_write.c

```diff
@@ -185,7 +185,7 @@ dds_return_t dds_write_impl (dds_writer *wr, const void * data, dds_time_t tstam
     if (!config.whc_batch)
       nn_xpack_send (wr->m_xp, false);
     ret = DDS_RETCODE_OK;
-  } else if (w_rc == ERR_TIMEOUT) {
+  } else if (w_rc == Q_ERR_TIMEOUT) {
```

Comment:
I can see the pragmatism behind this, but just renaming ERR_TIMEOUT to Q_ERR_TIMEOUT is a bit too pragmatic to my taste. I'd rather the DDSI stack started using the same DDS_RETCODE_xxx codes instead of defining its own ones.

Reply:
@eboasson, I had a feeling you'd say that. Agreed. I'll incorporate changing the `ERR_*` variables.

Reply:
On second thought, I'll skip this. Error codes in ddsi are negative and some are used in functions that return a positive integer for a count. To replace the `ERR_*` codes with `DDS_RETCODE_*` equivalents means that I need to change the `DDS_RETCODE_*` ones to be negative, which is a bit to much work to do simultaneously with making FreeRTOS a tier-1 platform.